### PR TITLE
fix: (Core) address accessibility issues for the Object marker component

### PR DESCRIPTION
**Pull request by Lokanathan-k**
fixing the Accessibility issue for the Object marker component

#### Please provide a link to the associated issue.

https://github.com/SAP/fundamental-ngx/issues/3558
https://github.com/SAP/fundamental-ngx/issues/3557
https://github.com/SAP/fundamental-ngx/issues/3821

#### Please provide a brief summary of this pull request.

fixed the tab index and Accessibility issues.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/object-marker/examples/object-marker-example.component.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
-<span fd-object-marker glyph="add-favorite" title="Favorite" aria-label="Favourite Icon" aria-hidden="true"></span>
-<span fd-object-marker glyph="flag" title="Flag" aria-label="Favourite Flag" aria-hidden="true"></span>
-<span fd-object-marker glyph="request" title="Request" aria-label="Favourite Request" aria-hidden="true"></span>
-<span fd-object-marker glyph="user-edit" title="Editable" aria-label="Favourite Editable" aria-hidden="true"></span>
-<span fd-object-marker glyph="private" title="Private" aria-label="Favourite Private" aria-hidden="true"></span>
+<span fd-object-marker glyph="add-favorite" title="Favorite" aria-label="Favorite Icon" aria-hidden="true"></span>
+<span fd-object-marker glyph="flag" title="Flag" aria-label="Favorite Flag" aria-hidden="true"></span>
+<span fd-object-marker glyph="request" title="Request" aria-label="Favorite Request" aria-hidden="true"></span>
+<span fd-object-marker glyph="user-edit" title="Editable" aria-label="Favorite Editable" aria-hidden="true"></span>
+<span fd-object-marker glyph="private" title="Private" aria-label="Favorite Private" aria-hidden="true"></span>
```
Comment (KevinOkamoto) [inline apps/docs/src/app/core/component-docs/object-marker/examples/object-marker-example.component.html:5]: I think you need to set `aria-hidden=false` or else the screen reader will skip these.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-hidden_attribute

**Review by KevinOkamoto — CHANGES_REQUESTED** [review]
I think you need to edit the title of this PR to be in compliance with guidelines. You don't have a "(scope)" in your title.

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,3 +1,3 @@
 {
     "editor.formatOnSave": false
-}
\ No newline at end of file
+}
```
Comment (salarenko) [inline .vscode/settings.json:3]: Revert

File: apps/docs/src/app/core/component-docs/object-marker/examples/object-marker-example.component.html [hunk 3]
```diff
@@ -1,5 +1,5 @@
-<span fd-object-marker glyph="add-favorite" title="Favorite" aria-label="Favourite Icon" aria-hidden="true"></span>
-<span fd-object-marker glyph="flag" title="Flag" aria-label="Favourite Flag" aria-hidden="true"></span>
-<span fd-object-marker glyph="request" title="Request" aria-label="Favourite Request" aria-hidden="true"></span>
-<span fd-object-marker glyph="user-edit" title="Editable" aria-label="Favourite Editable" aria-hidden="true"></span>
-<span fd-object-marker glyph="private" title="Private" aria-label="Favourite Private" aria-hidden="true"></span>
+<span fd-object-marker glyph="add-favorite" title="Favorite" aria-label="Favorite Icon" aria-hidden="false"></span>
```
Comment (salarenko) [inline apps/docs/src/app/core/component-docs/object-marker/examples/object-marker-example.component.html:1]: Why are you using `aria-hidden` here?

File: apps/docs/src/app/core/component-docs/object-marker/examples/object-marker-text-example.component.html [hunk 4]
```diff
@@ -1,4 +1,10 @@
...
-<a fd-object-marker clickable="true" aria-label="Object Maker with only Text Clickable" label="Locked"></a>
-<a fd-object-marker clickable="true" aria-label="Object Maker with only Text Clickable" label="Edit"></a>
+<span fd-object-marker aria-label="Object Marker with Text" label="Draft"></span>
+<span fd-object-marker aria-label="Object Marker with Text" label="Request"></span>
+<a
+    fd-object-marker
+    clickable="true"
+    aria-label="Object Marker with only Text Clickable"
+    label="Locked"
+    tabindex="0"
```
Comment (salarenko) [inline apps/docs/src/app/core/component-docs/object-marker/examples/object-marker-text-example.component.html:8]: Isn't this weird that you need to explicitly add tabindex? It should not be like that, anchor tag is tabbable by default.

**Review by salarenko — CHANGES_REQUESTED** [review]

**Review by Betrozov — APPROVED** [review]

**Review by salarenko — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Review by KevinOkamoto — APPROVED** [review]

**Pull request merged**